### PR TITLE
Session keys - UI components

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Does not close anything but it is part of #4413 

## Description of Changes
- Added new tag component that handles different cases (left icon, address/username, right icon)
- Added address connection indicator and button for joining the community
- Extended banner component to support all types

Pasted Graphic 6

Pasted Graphic 7

Pasted Graphic 4

Pasted Graphic 2

## Test Plan
- for now everything is hidden behind the `sessionKeys` feature flag so it is not possible to see components unless you set it to true on your local env
- components do not work, they will be wired up in the upcoming PRs

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx [hunk 1]
```diff
@@ -148,8 +150,24 @@ export const HeaderWithFilters = ({
...
+
+  const handleJoinCommunity = () => {
+    console.log('click handle join');
+  };
+
   return (
     <div className="HeaderWithFilters">
+      {featureFlags.sessionKeys && (
+        <SwitchAddressBanner
+          communityName="dydx"
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx:165]: Can we note somehow that this hardcoded string needs to change?

File: packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx [hunk 2]
```diff
@@ -148,8 +150,24 @@ export const HeaderWithFilters = ({
...
+          onCommunityJoin={handleJoinCommunity}
+          onAddressSwitch={handleSwitchAddress}
+          onClose={() => console.log('close')}
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx:168]: And this onClose.

File: packages/commonwealth/webpack/webpack.base.config.js [hunk 3]
```diff
@@ -60,6 +60,9 @@ module.exports = {
         process.env.FLAG_PROPOSAL_TEMPLATES
       ),
     }),
+    new webpack.DefinePlugin({
```
Comment (jnaviask) [inline packages/commonwealth/webpack/webpack.base.config.js:63]: Can we use the FLAG_ prefix for the environment var, to maintain consistency?

**Review by jnaviask — APPROVED** [review]
Should we write follow-up tickets to get the new components in storybook? Or are they too large / that would be a difficult task?

As for this PR, some small notes that aren't blocking merging.

File: packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx (hunk 1 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx:166]: added comment

File: packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx (hunk 2 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx:169]: added comment

File: packages/commonwealth/webpack/webpack.base.config.js (hunk 3 above)
Comment (masvelio) [reply packages/commonwealth/webpack/webpack.base.config.js:63]: nice catch, thanks!

**Comment by masvelio** [comment]
ready 2 go

**Pull request merged**